lib/helpers.php

```diff
@@ -149,7 +153,7 @@ function micropub_post($endpoint, $params, $access_token, $file_path = NULL, $js
     $mimetype = finfo_file($finfo, $file_path);
     $multipart = new p3k\Multipart();
     $multipart->addArray($params);
-    $multipart->addFile($file_prop, $file_path, $mimetype);
+    $multipart->addFile($file_prop, $filename, $mimetype, $file_content);
```

Comment:
You sure? It definitely seems according to the source. It has a $content=null as the 4th parameter and later checks that and behaves accordingly.

https://github.com/aaronpk/php-multipart-encoder/blob/master/src/p3k/Multipart.php#L41